make indent a prop

carlos22 opened the pull request:
This makes the indent a prop so it can be configured by the user of the lib. Default is still 18px.

lycHub commented:
I think it would be more convenient to use "css var", What do you think?
```
[`${props.indentType}Left`]: `calc(${props.node.level} * var(--vir-tree-indent, 18px))`
````

carlos22 commented:
Yeah that's also a very good idea and even a bit simpler. I'll update the PR.

carlos22 committed:
move indent calculation to css with variables to offer more flexibility

carlos22 commented:
I even went a step further and just added a level variable and moved the padding itself to the css file. This allows us to remove the `indentType` prop and offers more possibilities to style. It also makes it a bit more clear as we don't put any style directly on the element but only a variable. What do you think?

lycHub commented:
Can this default value be omitted so that users can set "-- vir-tree-indent" on any parent node, not just limited to ". vir tree".

![image](https://github.com/user-attachments/assets/08965a0e-31d0-4f03-99c3-e0b11e92d5f0)

carlos22 commented:
> Can this default value be omitted so that users can set "-- vir-tree-indent" on any parent node, not just limited to ". vir tree".
> 
> ![image](https://private-user-images.githubusercontent.com/29920553/362191077-08965a0e-31d0-4f03-99c3-e0b11e92d5f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4MzI5MDQsIm5iZiI6MTcyNDgzMjYwNCwicGF0aCI6Ii8yOTkyMDU1My8zNjIxOTEwNzctMDg5NjVhMGUtMzFkMC00ZjAzLTk5YzMtZTBiMTFlOTJkNWYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI4VDA4MTAwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE3MDE2NDZlMGNhZDYxY2FmMTY0M2Q4NjQzOGQ3YjUwZmU3NzExNzZlOTAzNzBiMmNhYzJjNDU0MjAzYTAxMmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cptEjupq4MqRu2TbKFkDEmbWeWmL195_MYwrRo7A8hU)

Yes that is possible, but the users then need to know that there is this variable, thats why I put it there... 
But that does not work with all other variables?

carlos22 commented:
Changed it.

lycHub commented:
> Changed it.

I changed it back and update to v0.0.13

carlos22 commented:
Ok also fine for me :) Thanks a lot!
